File: objc/Chapter2/BowTies/ColorAttributeTransformer.m
//
//  ColorAttributeTransformer.m
//  BowTies
//
//  Created by Jinwoo Kim on 6/26/21.
//

#import "ColorAttributeTransformer.h"
#import <UIKit/UIKit.h>

@implementation ColorAttributeTransformer

+ (NSArray<Class> *)allowedTopLevelClasses {
    return @[[UIColor class]];
}

+ (void)register {
    NSString *className = NSStringFromClass(self);
    NSValueTransformerName name = (NSValueTransformerName)className;
    
    ColorAttributeTransformer *transformer = [ColorAttributeTransformer new];
    [NSValueTransformer setValueTransformer:transformer forName:name];
}

@end

File: swift/Chapter2/BowTies/ColorAttributeTransformer.swift
//
//  ColorAttributeTransformer.swift
//  BowTies
//
//  Created by Jinwoo Kim on 6/21/21.
//  Copyright © 2021 Razeware. All rights reserved.
//

import UIKit

class ColorAttributeTransformer: NSSecureUnarchiveFromDataTransformer {
  // 1
  override static var allowedTopLevelClasses: [AnyClass] {
    return [UIColor.self]
  }
  
  // 2
  static func register() {
    let className = String(describing: ColorAttributeTransformer.self)
    let name = NSValueTransformerName(className)
    
    let transformer = ColorAttributeTransformer()
    ValueTransformer.setValueTransformer(transformer, forName: name)
  }
}
